draft-irtf-cfrg-rsa-blind-signatures.md

```diff
@@ -395,6 +395,9 @@ An alternative solution to this problem of message blindness is to give signers
 message being signed is well-structured. Depending on the application, zero knowledge proofs
 could be useful for this purpose. Defining such a proof is out of scope for this document.
 
+Verifiers should check that, in addition to the signature validity, the unblinded message is 
+well-structured.
```

Comment:
```suggestion
Verifiers should check that, in addition to signature validity, the unblinded message is 
well-structured for the relevant application. For example, if an application of this protocol
requires messages to be structures of a particular form, then verifiers should check that 
unblinded messages adhere to this form.
```

If the message is expected to be a well-defined message, e.g.:

```
struct {
    uint32 foo;
    uint8 bar;
} Baz;
```

The verifier could check that the message deserializes correctly. Maybe this suggestion clarifies?

Reply:
Yup this is far more clear than what I had. @FredericJacobs does this cover your question?